backend/main.py

```
@@ -9,46 +1,343 @@
+        # Query for illustration-related content
+        response = rag_system.query("Show me all illustrations and artwork")
+
+        return {
+            "message": "Illustrations are now auto-discovered! Use the /query endpoint.",
+            "query_example": "Ask: 'Show me Nick's illustrations' or 'What artwork does Nick have?'",
+            "response_preview": response[:200] + "..." if len(response) > 200 else response,
+        }
+
+    except Exception as e:
+        logger.error(f"❌ Illustrations query failed: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)